runelite-client/src/main/java/net/runelite/client/plugins/microbot/qualityoflife/scripts/WintertodtScript.java

```diff
@@ -0,0 +1,250 @@
+                        Microbot.getClientThread().invokeLater(() -> Rs2GameObject.interact(unlitBrazier, "Light"));
+                    }
+                }
+            }
+            helpedIncapitatedPyromancer = false;
+        }
+    }
+
+    public void onNpcSpawned(NpcSpawned event) {
+        if (event.getNpc().getId() == 7372) {
+            if (incapitatedPyromancer != null) {
+                if (incapitatedPyromancer.getWorldLocation().distanceTo2D(Rs2Player.getWorldLocation()) > event.getNpc().getWorldLocation().distanceTo2D(Rs2Player.getWorldLocation())) {
+                    incapitatedPyromancer = event.getNpc();
+                    return;
+                }
+            }
+            incapitatedPyromancer = event.getNpc();
+        }
+        if (event.getNpc().getId() == 7371) {
+            if (pyromancer != null) {
+                if (pyromancer.getWorldLocation().distanceTo2D(Rs2Player.getWorldLocation()) > event.getNpc().getWorldLocation().distanceTo2D(Rs2Player.getWorldLocation())) {
+                    pyromancer = event.getNpc();
+                    return;
+                }
+            }
+            pyromancer = event.getNpc();
+        }
+    }
+
+    public void onNpcDespawned(NpcDespawned event) {
+        if (event.getNpc().equals(incapitatedPyromancer)) {
+            incapitatedPyromancer = Microbot.getClientThread().runOnClientThread(() -> Rs2Npc.getNpc("Incapacitated Pyromancer"));
+        }
+        if (event.getNpc().equals(pyromancer)) {
+            pyromancer = Microbot.getClientThread().runOnClientThread(() -> Rs2Npc.getNpc("Pyromancer"));
+        }
+    }
+
+    public void onChatMessage(ChatMessage chatMessage) {
+        var message = chatMessage.getMessage();
+        if (message.contains("The brazier is broken and shrapnel")) {
+            brokenBrazier = Microbot.getClientThread().runOnClientThread(() -> Rs2GameObject.getGameObjects(ObjectID.BRAZIER_29313).stream().filter(gameObject -> gameObject.getWorldLocation().distanceTo2D(Rs2Player.getWorldLocation()) < 5).findFirst().orElse(null));
+            Microbot.log("Broken brazier: " + brokenBrazier);
+            if (config.fixBrokenBrazier()) {
+                qolPlugin.updateWintertodtInterupted(false);
+                Global.sleepUntilOnClientThread(() -> (brokenBrazier != null && brokenBrazier.getWorldLocation().distanceTo2D(Rs2Player.getWorldLocation()) < 5), 1000);
+                Microbot.getClientThread().invokeLater(() -> Rs2GameObject.interact(brokenBrazier, "Fix"));
+            }
+        }
+
+        if (message.startsWith("The brazier has gone out")) {
+            unlitBrazier = Microbot.getClientThread().runOnClientThread(() -> Rs2GameObject.getGameObjects(ObjectID.BRAZIER_29312).stream().filter(gameObject -> gameObject.getWorldLocation().distanceTo2D(Rs2Player.getWorldLocation()) < 5).findFirst().orElse(null));
```

Comment:
Isn't this code already running on the client thread? I don't know how I feel about running Rs2Utility classes on the main thread..This can cause issues, because under the hood we have mouse clicks that can sleep between clicks.

Reply:
Oversight by me while i was doing some troubleshooting, i will edit the code